<!-- Phidgets IMU launch file -->

<launch>

    <node pkg="nodelet" type="nodelet" name="pimu_manager" 
        args="manager" output="screen">
	</node>

    <node pkg="nodelet" type="nodelet" name="PhidgetsImuNodelet" 
        args="load phidgets_imu/PhidgetsImuNodelet pimu_manager" 
        output="screen">

		<param name="frame_id" value="pimu"/>
		<remap from="/imu/data_raw" to="/pimu/data_raw"/>
		<remap from="/imu/mag" to="/pimu/mag"/>
		<remap from="/imu/is_calibrated" to="/pimu/is_calibrated"/>

        <!-- supported data rates: 4 8 16 24 32 40 ... 1000 (in ms) -->
        <param name="period" value="4"/>

        <!-- compass correction params (see http://www.phidgets.com/docs/1044_User_Guide) -->
        <param name="cc_mag_field" value="0.170291"/>
        <param name="cc_offset0" value="-0.060594"/>
        <param name="cc_offset1" value="-0.284538"/>
        <param name="cc_offset2" value="0.000000"/>
        <param name="cc_gain0" value="3.143669"/>
        <param name="cc_gain1" value="8.600936"/>
        <param name="cc_gain2" value="5.872303"/>
        <param name="cc_t0" value="-0.001145"/>
        <param name="cc_t1" value="0.000000"/>
        <param name="cc_t2" value="-0.000419"/>
        <param name="cc_t3" value="0.000000"/>
        <param name="cc_t4" value="0.000000"/>
        <param name="cc_t5" value="0.000000"/>

        <!--<param name="serial_number" value="354968"/>-->

    </node>

<node pkg="nodelet" type="nodelet" name="ImuFilterNodelet" 
    args="load imu_filter_madgwick/ImuFilterNodelet pimu_manager" 
    output="screen">

    <param name="world_frame" value="enu"/>
    <param name="use_mag" value="true"/>

	<remap from="/imu/data_raw" to="/pimu/data_raw"/>
	<remap from="/imu/mag" to="/pimu/mag"/>
    <remap from="/imu/data" to="/pimu"/>

    <param name="publish_tf" value="false"/>
    <param name="orientation_stddev" value="0.01"/>

</node>

</launch>
